File: CINS237 Projects/Main/Character Sequence Challenge.vb
Module Character_Sequence_Challenge
    Sub main()
        System.Console.WriteLine(getMode("abcdefghgggijklaaaaayurr"))
    End Sub

    Public Function getMode(ByVal input As String) As Integer
        Dim previousChar As Char = Nothing
        Dim occurances As Integer = 0
        Dim V As Integer = 0
        For i As Integer = 0 To input.Length - 1
            'checks if the current char is the last char analyzed
            If input.Substring(i, 1) = previousChar Then
                occurances += 1
            Else
                'this character is different than the last character
                ' sets the new character to chount
                previousChar = input.Substring(i, 1)
                'sets the largest occured char as the occurances of that char
                If occurances > V Then
                    V = occurances
                End If
                'reset the number of chars to 1, to analyze the next char
                occurances = 1
            End If
        Next
        Return V
    End Function
End Module

File: CINS237 Projects/_GUI Projects CSharp/Character Sequence Challenge.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace _GUI_Projects_CSharp
{
    class Character_Sequence_Challenge
    {
        public static void main()
        {
            System.Console.WriteLine(getOccurences("abcdefghgggijklaaaaayurr"));
        }

        public static int getOccurences(string input)
        {
            string previousChar = "";
            int occurences = 0;
            int V = 0;
            for (int i = 0; i <= input.Length - 1; i++)
            {
                if (input.Substring(i,1) == previousChar)  occurences++;
                else
                {
                    previousChar = input.Substring(i, 1);
                    if ( occurences > V) 
                        V=occurences;
                    occurences = 1;
                }
            }
            return V;
        }
    }
}
